src/js/hooks/cycleCount/useCheckStep.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import { createColumnHelper } from '@tanstack/react-table';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import cycleCountApi from 'api/services/CycleCountApi';
+import { TableCell } from 'components/DataTable';
+import TableHeaderCell from 'components/DataTable/TableHeaderCell';
+import { CYCLE_COUNT } from 'consts/applicationUrls';
+import useTranslate from 'hooks/useTranslate';
+
+const useCheckStep = () => {
+  const tableData = useRef([]);
+  const [dateCounted, setDateCounted] = useState({});
+  const history = useHistory();
+  const columnHelper = createColumnHelper();
+  const translate = useTranslate();
+
+  const {
+    cycleCountIds,
+    currentLocation,
+  } = useSelector((state) => ({
+    cycleCountIds: state.cycleCount.toCount,
+    currentLocation: state.session.currentLocation,
+  }));
+
+  useEffect(() => {
+    (async () => {
+      const { data } = await cycleCountApi.getCycleCounts(
+        currentLocation?.id,
+        cycleCountIds,
+      );
+      tableData.current = data?.data;
+      const countedDates = data?.data?.reduce((acc, cycleCount) => ({
+        ...acc,
+        [cycleCount?.id]: cycleCount?.cycleCountItems[0].dateCounted,
+      }), {});
+      setDateCounted(countedDates);
+    })();
+  }, [cycleCountIds, currentLocation]);
+
+  const back = () => {
+    history.push(CYCLE_COUNT.countStep());
+  };
+
+  const save = () => {
+    console.log('save');
+  };
+
+  const getCountedDate = (cycleCountId) => dateCounted[cycleCountId];
+
+  const columns = [
```

Comment:
isn't there `useMemo` missing? I know there were problems with translations, but we need to figure out the bug, instead of getting rid of the memoization.